airflow/www/views.py

```diff
@@ -181,6 +181,11 @@ def sanitize_args(args: dict[str, str]) -> dict[str, str]:
 )
 
 
+def _sorted_groupby(it, key: Callable | None = None, reverse: bool = False):
+    """Helper around `itertools.groupby`, which pre-sort iterable by the same key."""
+    yield from itertools.groupby(sorted(it, key=key, reverse=reverse), key=key)
```

Comment:
I’d put this function closer to its usage, maybe nested in `duration` directly. This is more than 3,000 lines apart!

Reply:
Got it. Lets return to initial suggestion, to move into nested function for determine grouping key